### PR TITLE
feat(app): create well selection component

### PR DESCRIPTION
**Pull request by brenthagen**
# Overview

creates ODD well selection component for use in quick transfer. implements basic source and destination quick transfer well selection. based on `SelectableLabware` component in protocol designer.

closes PLAT-172, PLAT-175

TODO: tweak styling: selection rectangle layout, well labels
TODO: refactor labware render component to simplify Well component rendering and remove SVG well stroke

Screen Shot 2024-05-16 at 4 03 30 PM

# Test Plan

verified touch selection on device

# Changelog

 - Creates well selection component
 - Implements quick transfer source and destination well selection

# Review requests

select some wells

# Risk assessment

low

### CODE REVIEW

File: app/src/organisms/WellSelection/index.tsx [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+import type { WellFill, WellGroup, WellStroke } from '@opentrons/components'
+import type {
+  LabwareDefinition2,
+  PipetteChannels,
+} from '@opentrons/shared-data'
+import type { GenericRect } from './types'
+
+interface WellSelectionProps {
+  definition: LabwareDefinition2
+  /** array of primary wells. Overrides labwareProps.selectedWells */
```
Comment (b-cooper) [inline app/src/organisms/WellSelection/index.tsx]: is this comment still accurate?

Comment (brenthagen) [reply app/src/organisms/WellSelection/index.tsx]: since we're no longer spreading labware props, no

**Review by smb2268 — APPROVED** [review]
this looks great! I tested it out on the dev app with a single, 8 and 96-channel and all worked as expected and wells are getting added to state properly.

**Review by b-cooper — APPROVED** [review]
changes look good ☑️

**Pull request merged**